assets/js/modules/search-console/datastore/report.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * modules/search-console data store: report.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { stringifyObject } from '../../../util';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { isValidDateRange } from '../../../util/report-validation';
+
+const fetchGetReportStore = createFetchStore( {
+	baseName: 'getReport',
+	controlCallback: ( { options } ) => {
+		return API.get( 'modules', 'search-console', 'searchanalytics', options );
+	},
+	reducerCallback: ( state, report, { options } ) => {
+		return {
+			...state,
+			reports: {
+				...state.reports,
+				[ stringifyObject( options ) ]: report,
+			},
+		};
+	},
+	argsToParams: ( options ) => {
+		invariant( isPlainObject( options ), 'Options for Analytics report must be an object.' );
+		invariant( isValidDateRange( options ), 'Either date range or start/end dates must be provided for Analytics report.' );
```

Comment:
These strings should use "Search Console" :)

Reply:
Updated :)